h/api/models/elastic.py

```diff
@@ -117,7 +117,7 @@ def shared(self):
         elif len(read_perms) > 1 and self.groupid == '__world__' and gperm in read_perms:
             read_perms = [gperm]
 
-        if (read_perms != [gperm] and read_perms != [self.userid]):
+        if (gperm not in read_perms and read_perms != [self.userid]):
```

Comment:
I didn't realise that the function was supposed to be as conservative as possible in its interpretation.

The reason for this change is that the existing code will match:

```
["group:wibble"]
```

but will _not_ match:

```
["group:wibble", "acct:seanh@hypothes.is"]
```

as being shared, which of course isn't a _correct_ interpretation of that ACL. And the `reply_template` tests do use `["group:wibble", "acct:seanh@hypothes.is"]`-type read ACLs in some global shared fixtures and expect those annotations to be treated as shared or the tests fail.

The alternative would be to fix those tests, either to just use `["group:wibble"]` or to patch the model code.